ghcide/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -137,48 +139,49 @@ runLanguageServer options inH outH getHieDbLoc defaultConfig onConfigurationChan
             registerIdeConfiguration (shakeExtras ide) initConfig
 
             let handleServerException (Left e) = do
-                    logError (ideLogger ide) $
+                    logError logger $
                         T.pack $ "Fatal error in server thread: " <> show e
                     exitClientMsg
                 handleServerException _ = pure ()
+
+                uncaughtError (e :: SomeException) = do
+                    logError logger $ T.pack $
+                        "Unexpected exception on notification, please report!\n" ++
```

Comment:
```suggestion
                        "Unexpected exception or notification, please report!\n" ++
```

Reply:
Generally, is there a reason why the log message and the window message are different here?

Reply:
This function is called from multiple places, not just the notification handler.